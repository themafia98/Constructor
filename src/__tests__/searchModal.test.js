import React from 'react';
import renderer from 'react-test-renderer';
import SearchModal from '../components/ModalWindow/Search/SearchModal';
import Events from 'events';


const cancelTest = event => {
    console.log('cancelTest work search modal!');
}

const cbShowImage = event => {
    console.log('cbShowImage work search modal!');
}

const cbSetSelectedImage = event => {
    console.log('cbSetSelectedImage work search modal!');
}

const cbSearch = event => {
    console.log('cbSearch work search');
}
    let stream2 = new Events();

    const ModalWindowComponent = renderer.create(
        <SearchModal
            modalSearchMode = 'image'
            content = {{}}
            cbSearch  = {cbSearch}
            cbCancel = {cancelTest}
            cbShowImage = {cbShowImage}
            cbSetSelectedImage = {cbSetSelectedImage}
        />
    );

    test("SearchModal test", () =>{

        let snapshotModalWindowComponent = ModalWindowComponent.toJSON();
        expect(snapshotModalWindowComponent).toMatchSnapshot();


        const projectSelect = ModalWindowComponent.root.find(item => item.props.className === 'acceptButton');
        projectSelect.props.onClick({stopPropagation: () => null});

        snapshotModalWindowComponent = ModalWindowComponent.toJSON();
        expect(snapshotModalWindowComponent).toMatchSnapshot();
    });