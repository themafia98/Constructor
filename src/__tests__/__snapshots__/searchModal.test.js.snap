// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchModal test 1`] = `
<div
  className="Modal Modal-search"
>
  <h3>
    Search image
  </h3>
  <div
    className="Modal-search__controllers"
  >
    <input
      type="text"
    />
    <input
      className="acceptButton"
      onClick={[Function]}
      type="button"
      value="Search"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Cancel"
    />
  </div>
  <div
    className="searchResultBox"
  />
</div>
`;

exports[`SearchModal test 2`] = `
<div
  className="Modal Modal-search"
>
  <h3>
    Search image
  </h3>
  <div
    className="Modal-search__controllers"
  >
    <input
      type="text"
    />
    <input
      className="acceptButton"
      onClick={[Function]}
      type="button"
      value="Search"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Cancel"
    />
  </div>
  <div
    className="searchResultBox"
  />
</div>
`;
